example/lib/entities/rectangle/behaviors/circle_collision_behavior.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:example/entities/entities.dart';
+import 'package:flame/extensions.dart';
+import 'package:flame_behaviors/flame_behaviors.dart';
+import 'package:flutter/material.dart';
+
+class CircleCollisionBehavior extends CollisionBehavior<Circle, Rectangle> {
+  final _collisionColor = Colors.green.withOpacity(0.8);
+
+  @override
+  void onCollisionStart(Set<Vector2> intersectionPoints, Circle other) {
+    parent.paint.color = _collisionColor;
+  }
+
+  @override
+  void onCollisionEnd(Circle other) {
+    if (!isColliding) {
```

Comment:
Curious what would be an example of where `onCollisionEnd` is called but `isColliding` is `true`?

Reply:
The `isColliding` indicates that we are still colliding with other entities(or the same one) of the given `Collider` type. So we leave collision with one, we could still have one with another.

Reply:
ah got it -- `isColliding` is not scoped to the current collision, makes sense 👍